packages/backend/src/Controllers/Roll/addElectionRollController.ts

```diff
@@ -23,6 +23,15 @@ const className = "VoterRolls.Controllers";
 const addElectionRoll = async (req: IElectionRequest & { body: { electionRoll: ElectionRollInput[] } }, res: Response, next: NextFunction) => {
     expectPermission(req.user_auth.roles, permissions.canAddToElectionRoll)
     Logger.info(req, `${className}.addElectionRoll ${req.election.election_id}`);
+
+    // Prevent creating voters by voter_id when using email invitations
+    if (req.election.settings.invitation === 'email') {
+        const hasVoterId = req.body.electionRoll.some((rollInput: ElectionRollInput) => rollInput.voter_id);
+        if (hasVoterId) {
+            throw new BadRequest('Cannot create voters with voter_id when using email invitations');
+        }
+    }
```

Comment:
Heads-up: there is going to be merge conflict here bc I also made this change and pushed it